stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -791,7 +793,9 @@ public class ChatClient internal constructor(
     private fun getTokenAndConnect(connect: () -> Unit) {
         tokenManager.loadAsync {
             if (it.isSuccess) {
-                state.token = it.data()
+                clientStateService.onTokenReceived(it.data())
+            } else {
+                clientStateService.onTokenReceived("")
```

Comment:
We have such strong typing everywhere else, can we avoid using an empty string to indicate the failure? Could we have a separate method, something like `clientStateService.onTokenLoadFailed()`?

Another question: do we handle the special case of the empty string somewhere later on in our code?

Reply:
Marton, your comment is right. But this hasn't been implemented before and I don't know how to handle this case. I use empty token as it worked before.

Reply:
@JcMinarro What do you think we suppose to do if we failed in token getting?

Reply:
Would be good to have a error state on that case, and doesn't try to connect (It is not in a good state to be connected)

Reply:
I made the standalone [Jira task for that ](https://stream-io.atlassian.net/browse/CAS-437?atlOrigin=eyJpIjoiMWFjOTU2MTVlMTdiNGM2OTgyMmM4MTJjYmQxZTk0ZGUiLCJwIjoiaiJ9)